File: Leetcode/19.removeNthNodeFromEndOfList/19.cpp
ListNode* removeNthFromEnd(ListNode* head, int n) 
{
	ListNode newhead(-1);
	ListNode * fast = &newhead;
	newhead.next = head;
	ListNode * slow = &newhead;
	while (n > 0)
	{
		fast = fast -> next;
		--n;
	}
	while (fast -> next)
	{
		fast = fast -> next;
		slow = slow -> next;
	}
	slow -> next = slow -> next -> next;
	return newhead.next;
}

File: Leetcode/19.removeNthNodeFromEndOfList/19.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        fast = slow = head
        for _ in range(n): # _ it's just a variable like any other, but by convention it means that you don't intend
        #to use that value, just read it and ignore it
        	fast = fast.next
        if not fast:
        	return head.next
        while fast.next:
        	fast = fast.next
        	slow = slow.next
        slow.next = slow.next.next
        return head
        
